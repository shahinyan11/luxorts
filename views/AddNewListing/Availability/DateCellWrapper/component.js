import classNames from 'classnames';
import PropTypes from 'prop-types';

import useContainer from './hook';

const DateCellWrapper = ({ value, ...props }) => {
  const { isToday, isActual, isBlocked, isAnotherMonth, isSelectedCell } = useContainer({
    value,
    ...props,
  });

  return (
    <div
      className={classNames('rbc-day-bg', {
        'rbc-off-range-bg': isAnotherMonth,
        'rbc-selected-cell': isSelectedCell && isActual && !isAnotherMonth,
      })}
    >
      <div
        className={classNames('aviability-calendar__event', {
          'aviability-calendar__event--today': isToday,
          'aviability-calendar__event--overdue': !isActual,
          'aviability-calendar__event--blocked': isBlocked && !isAnotherMonth,
        })}
      >
        {!isAnotherMonth && <p className="aviability-calendar__counter mb-0">{value.getDate()}</p>}
      </div>
    </div>
  );
};

DateCellWrapper.propTypes = {
  value: PropTypes.instanceOf(Date).isRequired,
  children: PropTypes.node.isRequired,
  listingBlockedTimePeriods: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default DateCellWrapper;
