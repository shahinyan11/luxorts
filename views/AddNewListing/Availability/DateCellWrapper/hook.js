import { compose, includes, pathOr } from 'ramda';

import { DATE_FORMAT } from 'constants';

import checkDateInfo from 'utils/checkDateInfo';

function useContainer({ value, children, listingBlockedTimePeriods }) {
  const { isActual, isToday, momentDate } = checkDateInfo(value);

  // calculate cell day state
  const isBlocked = includes(momentDate.format(DATE_FORMAT), listingBlockedTimePeriods);
  const isAnotherMonth = compose(
    includes('rbc-off-range-bg'),
    pathOr('', ['props', 'className']),
  )(children);
  const isSelectedCell = compose(
    includes('rbc-selected-cell'),
    pathOr('', ['props', 'className']),
  )(children);

  return {
    isActual,
    isBlocked,
    isToday,
    isAnotherMonth,
    isSelectedCell,
  };
}

export default useContainer;
