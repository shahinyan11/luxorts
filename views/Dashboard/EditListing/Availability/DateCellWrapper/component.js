import classNames from 'classnames';
import PropTypes from 'prop-types';

import { DATE_FORMAT } from 'constants';

import checkDateInfo from 'utils/checkDateInfo';

const DateCellWrapper = ({ value, pricePerDay, customPrice, children }) => {
  const { momentDate, isActual } = checkDateInfo(value);
  const dateString = momentDate.format(DATE_FORMAT);
  const price = `$${customPrice[dateString] || pricePerDay}`;

  return (
    <div
      className={classNames(children.props.className, {
        'rbc-day-bg--overdue': !isActual,
      })}
    >
      {price}
    </div>
  );
};

DateCellWrapper.propTypes = {
  value: PropTypes.instanceOf(Date).isRequired,
  children: PropTypes.node.isRequired,
  pricePerDay: PropTypes.number.isRequired,
  customPrice: PropTypes.shape().isRequired,
};

export default DateCellWrapper;
