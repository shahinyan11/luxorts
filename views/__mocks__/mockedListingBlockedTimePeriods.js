export default ['25/07/2022', '26/07/2022', '27/07/2022'];
