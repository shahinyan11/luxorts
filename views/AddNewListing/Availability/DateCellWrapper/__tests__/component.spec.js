import { shallow } from 'enzyme';

import mockedListingBlockedTimePeriods from 'views/__mocks__/mockedListingBlockedTimePeriods';

import useContainer from '../hook';
import DateCellWrapper from '../component';

const mockedHookData = {
  isToday: true,
  isActual: true,
  isBlocked: false,
  isAnotherMonth: false,
  isSelectedCell: false,
};
jest.mock('../hook', () => jest.fn(() => mockedHookData));

describe('DateCellWrapper component', () => {
  const props = {
    value: new Date('15 Jun 2022 00:00:00 GMT'),
    listingBlockedTimePeriods: mockedListingBlockedTimePeriods,
  };

  let component = shallow(<DateCellWrapper {...props}>Foo</DateCellWrapper>);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isAnotherMonth equals true and isToday equals false', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      isToday: false,
      isAnotherMonth: true,
    });

    component = shallow(<DateCellWrapper {...props}>Foo</DateCellWrapper>);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isToday equals false and isActual equals false', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      isToday: false,
      isActual: false,
    });

    component = shallow(<DateCellWrapper {...props}>Foo</DateCellWrapper>);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isBlocked equals true and isToday equals false', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      isToday: false,
      isBlocked: true,
    });

    component = shallow(<DateCellWrapper {...props}>Foo</DateCellWrapper>);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isSelectedCell equals true and isActual equals true and isAnotherMonth equals false', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      isSelectedCell: true,
    });

    component = shallow(<DateCellWrapper {...props}>Foo</DateCellWrapper>);

    expect(component).toMatchSnapshot();
  });
});
