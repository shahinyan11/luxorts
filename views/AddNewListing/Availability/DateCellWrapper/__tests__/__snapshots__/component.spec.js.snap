// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateCellWrapper component matches snapshot 1`] = `
<div
  className="rbc-day-bg"
>
  <div
    className="aviability-calendar__event aviability-calendar__event--today"
  >
    <p
      className="aviability-calendar__counter mb-0"
    >
      15
    </p>
  </div>
</div>
`;

exports[`DateCellWrapper component matches snapshot when isAnotherMonth equals true and isToday equals false 1`] = `
<div
  className="rbc-day-bg rbc-off-range-bg"
>
  <div
    className="aviability-calendar__event"
  />
</div>
`;

exports[`DateCellWrapper component matches snapshot when isBlocked equals true and isToday equals false 1`] = `
<div
  className="rbc-day-bg"
>
  <div
    className="aviability-calendar__event aviability-calendar__event--blocked"
  >
    <p
      className="aviability-calendar__counter mb-0"
    >
      15
    </p>
  </div>
</div>
`;

exports[`DateCellWrapper component matches snapshot when isSelectedCell equals true and isActual equals true and isAnotherMonth equals false 1`] = `
<div
  className="rbc-day-bg rbc-selected-cell"
>
  <div
    className="aviability-calendar__event aviability-calendar__event--today"
  >
    <p
      className="aviability-calendar__counter mb-0"
    >
      15
    </p>
  </div>
</div>
`;

exports[`DateCellWrapper component matches snapshot when isToday equals false and isActual equals false 1`] = `
<div
  className="rbc-day-bg"
>
  <div
    className="aviability-calendar__event aviability-calendar__event--overdue"
  >
    <p
      className="aviability-calendar__counter mb-0"
    >
      15
    </p>
  </div>
</div>
`;
